### DISCUSSION TITLE
Optional Dependencies

### DISCUSSION THREAD

**Discussion by pyodude** [pyodide/pyodide#1527] [Q&A]
While trying to use Plotly express, I got an error stating that pandas is required.

This got me to a (shallow) rabbit hole about python dependency management, and I discovered optional requirements :)

https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/optional-requirements.txt

Should we have pandas as a dependency for plotly in meta.yml (I can create a PR, as this is minimal), or leave that to the user, as seems to be normal approach (import pandas if you want to use plotly.express) ?

**Comment by xhluca** [comment]
I can't answer in the context of `pyodide`, but you might have users that want to use `plotly.express` without using `pandas`, e.g. passing in a dictionary and assign a key as `x`, `y`, `color`, etc.. Furthermore, the non-express plotly API doesn't directly use pandas. Adding it as a dependency would simply make the loading time for end user longer since you will need to download the `pandas.data` file which has a non-trivial size.
